src/data/polynomial/laurent.lean

```diff
@@ -365,12 +366,43 @@ instance (R : Type*) [semiring R] : is_scalar_tower R[X] R[X] R[T;T⁻¹] :=
 end semiring
 
 section comm_semiring
+variable [comm_semiring R]
 
 instance algebra_polynomial (R : Type*) [comm_semiring R] : algebra R[X] R[T;T⁻¹] :=
 { commutes' := λ f l, by simp [mul_comm],
   smul_def' := λ f l, rfl,
   .. polynomial.to_laurent }
 
+lemma algebra_map_X_pow (n : ℕ) : algebra_map R[X] R[T;T⁻¹] (X ^ n) = T n :=
+polynomial.to_laurent_X_pow n
+
+@[simp]
+lemma algebra_map_eq_to_laurent (f : R[X]) : algebra_map R[X] R[T;T⁻¹] f = f.to_laurent :=
+rfl
+
+lemma is_localization : is_localization (submonoid.closure ({X} : set R[X])) R[T;T⁻¹] :=
```

Comment:
It might be nicer to talk about the powers of `X` rather than the closure of `{X}`

Reply:
Do you mean using `powers_hom`?

Reply:
@eric-wieser what do you think? LGTM.